swap_meet/vendor.py

```diff
@@ -0,0 +1,89 @@
+from swap_meet.item import Item 
+
+class Vendor:
+
+    def __init__(self, inventory=None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory 
+
+    def add(self, new_item):
+        self.inventory.append(new_item)
+        return new_item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    
+    def get_by_category(self, category):
+        items_in_category = []
+        for item in self.inventory:
+            if item.category == category:
+                items_in_category.append(item)
+        return items_in_category
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.remove(my_item)
+            other_vendor.add(my_item)
+            other_vendor.remove(their_item)
+            self.add(their_item)
+            return True
+        else:
+            return False
+    
+    def swap_first_item(self, other_vendor):        
+        if len(self.inventory) == 0 or len(other_vendor.inventory) == 0:
```

Comment:
You can also do this this way because `[]` is falsey.

```suggestion
        if not self.inventory or not other_vendor.inventory:
```